gapir/client/client.go

```diff
@@ -32,84 +37,195 @@ type tyLaunchArgsKey string
 // line arguments when launching GAPIR. The property must be of type []string.
 const LaunchArgsKey tyLaunchArgsKey = "gapir-launch-args"
 
+type clientInfo struct {
+	device               bind.Device
+	arch                 device.Architecture
+	abi                  *device.ABI
+	deviceConnectionInfo deviceConnectionInfo
+	connection           gapir.Connection
+	bgConnection         *backgroundConnection
+}
+
+type deviceArch struct {
+	device bind.Device
+	arch   device.Architecture
+}
+
+// ConnectionKey is used by manager to obtain a connection
+type ConnectionKey deviceArch
+
 // Client is interface used to connect to GAPIR instances on devices.
 type Client struct {
-	mutex    sync.Mutex
-	sessions map[deviceArch]*session
+	mutex       sync.Mutex
+	clientInfos map[ConnectionKey]clientInfo
 }
 
+const (
+	connectTimeout    = time.Second * 10
+	heartbeatInterval = time.Millisecond * 500
+)
```

Comment:
move these const declaration to the top, merge it with the declaration of `LaunchArgsKey`